readthedocs/projects/views/private.py

```diff
@@ -455,47 +461,49 @@ class ProjectRelationshipDelete(ProjectRelationshipMixin, DeleteView):
     http_method_names = ['post']
 
 
-@login_required
-def project_users(request, project_slug):
-    """Project users view and form view."""
-    project = get_object_or_404(
-        Project.objects.for_admin_user(request.user),
-        slug=project_slug,
-    )
+class ProjectUsersMixin(ProjectAdminMixin, PrivateViewMixin):
 
-    form = UserForm(data=request.POST or None, project=project)
+    form_class = UserForm
 
-    if request.method == 'POST' and form.is_valid():
+    def get_queryset(self):
+        project = self.get_project()
+        return project.users.all()
+
+    def get_success_url(self):
+        return reverse('projects_users', args=[self.get_project().slug])
+
+
+class ProjectUsersCreateList(ProjectUsersMixin, FormView):
```

Comment:
Why we don't use `CreateView` for these?

Reply:
Because this view does two things, list and create. `CreatView` makes this a model view, we don't want that here.